frontend/src/metabase/querying/filters/hooks/use-filter-modal/utils/filters.unit.spec.ts

```diff
@@ -0,0 +1,100 @@
+import * as Lib from "metabase-lib";
+import { createQuery, createQueryWithClauses } from "metabase-lib/test-helpers";
+import { PRODUCTS_ID } from "metabase-types/api/mocks/presets";
+
+import { getGroupItems, hasFilters, removeFilters } from "./filters";
+
+const STAGE_COUNT = 4;
+
+function createFilteredQuery(query: Lib.Query) {
+  const joinTable = Lib.tableOrCardMetadata(query, PRODUCTS_ID);
+  const queryWithJoin = Lib.join(
+    query,
+    -1,
+    Lib.joinClause(
+      joinTable,
+      [
+        Lib.joinConditionClause(
+          query,
+          -1,
+          Lib.joinConditionOperators(query, -1)[0],
+          Lib.joinConditionLHSColumns(query, -1)[0],
+          Lib.joinConditionRHSColumns(query, -1, joinTable)[0],
+        ),
+      ],
+      Lib.availableJoinStrategies(query, -1)[0],
+    ),
+  );
+
+  const queryWithFilter = Lib.filter(
+    queryWithJoin,
+    -1,
+    Lib.expressionClause("=", [1, 1]),
+  );
+
+  return createQueryWithClauses({
+    query: queryWithFilter,
+    aggregations: [{ operatorName: "count" }],
+    breakouts: [{ tableName: "ORDERS", columnName: "TOTAL" }],
+  });
+}
+
+function createMultiStageFilteredQuery() {
+  const stageIndexes = Array.from({ length: STAGE_COUNT }, (_, i) => i);
+  return stageIndexes.reduce(
+    query => Lib.appendStage(createFilteredQuery(query)),
+    createQuery(),
+  );
+}
+
+describe("getGroupItems", () => {
+  it("should return groups for all stages", () => {
+    const query = createMultiStageFilteredQuery();
+    expect(getGroupItems(query).map(group => group.displayName)).toEqual([
+      "Orders",
+      "Products",
+      "User",
+      "Summaries",
+      "Products",
```

Comment:
Joins were added to test that we don't add `(2)`, `(3)` suffixes to non-main groups accidentally.